client/state/posts/counts/reducer.js

```diff
@@ -55,25 +62,124 @@ export function requesting( state = {}, action ) {
  * @param  {Object} action Action payload
  * @return {Object}        Updated state
  */
-export function counts( state = {}, action ) {
-	switch ( action.type ) {
-		case POST_COUNTS_RECEIVE:
-			return merge( {}, state, {
-				[ action.siteId ]: {
-					[ action.postType ]: action.counts
-				}
-			} );
+export const counts = ( () => {
+	let currentUserId;
+	let postStatuses = {};
 
-		case DESERIALIZE:
-			if ( isValidStateWithSchema( state, countsSchema ) ) {
-				return state;
+	/**
+	 * Returns a serialized key to be used in tracking post status properties
+	 *
+	 * @param  {Number} siteId Site ID
+	 * @param  {Number} postId Post ID
+	 * @return {String}        Serialized key
+	 */
+	function getPostStatusKey( siteId, postId ) {
+		return [ siteId, postId ].join();
```

Comment:
>to prevent a possible edge case ( very very very unlikely ) where a duplicate key could be created

Granted it's very unobvious, but the default separator for `join` is a comma, so there shouldn't be a worry of conflicts:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/join